account_bank_statement_import_online_ponto/models/online_bank_statement_provider_ponto.py

```diff
@@ -0,0 +1,244 @@
+        page_next = True
+        last_identifier = self.ponto_last_identifier
+        if last_identifier:
+            params['before'] = last_identifier
+            page_next = False
+        transaction_lines = []
+        latest_identifier = False
+        while page_url:
+            response = requests.get(page_url, verify=False, params=params,
+                                    headers=self._ponto_header())
+            if response.status_code == 200:
+                if params.get('before'):
+                    params.pop('before')
+                data = json.loads(response.text)
+                links = data.get('links', {})
+                if page_next:
+                    page_url = links.get('next', False)
+                else:
+                    page_url = links.get('prev', False)
+                transactions = data.get('data', [])
+                if transactions:
+                    current_transactions = []
+                    for transaction in transactions:
+                        date = self._ponto_date_from_string(
+                            transaction.get('attributes', {}).get('executionDate'))
+                        if date_since <= date < date_until:
+                            current_transactions.append(transaction)
+
+                    if current_transactions:
+                        if not page_next or (page_next and not latest_identifier):
+                            latest_identifier = current_transactions[0].get('id')
+                        transaction_lines.extend(current_transactions)
+            else:
+                raise UserError(
+                    _('Error during get transaction.\n\n%s \n\n %s') % (
+                        response.status_code, response.text))
+        if latest_identifier:
+            self.ponto_last_identifier = latest_identifier
+        return transaction_lines
+
+    def _ponto_date_from_string(self, date_str):
+        return datetime.strptime(date_str, '%Y-%m-%dT%H:%M:%S.%fZ')
+
+    def _ponto_obtain_statement_data(self, date_since, date_until):
+        self.ensure_one()
+        StatementLine = self.env['account.bank.statement.line']
+        Statement = self.env['account.bank.statement']
+
+        account_ids = self._ponto_get_account_ids()
+        journal = self.journal_id
+
+        iban = self.account_number
+        account_id = account_ids.get(iban)
+        if not account_id:
+            raise UserError(
+                _('Ponto : wrong configuration, unknow account %s')
+                % journal.bank_account_id.acc_number)
+
+        self._ponto_synchronisation(account_id)
+
+        transaction_lines = self._ponto_get_transaction(account_id,
+                                                        date_since,
+                                                        date_until)
+
+        new_transactions = []
+        sequence = 0
+        for transaction in transaction_lines:
+            if StatementLine.sudo().search(
```

Comment:
This filtering is done in base module, and import ID is prefixed always different